### ISSUE TITLE
Thoughts on best way to support preference and search_type

### ISSUE THREAD

**Issue by spiritinlife** [babenkoivan/elastic-adapter#18]
As per [elastic docs for consistent scoring ](https://www.elastic.co/guide/en/elasticsearch/reference/master/consistent-scoring.html), we should offer a way to set `search_type` and `preference` for a search operation.

This could be implemented to be done globally for all search operations and/or per search request. @babenkoivan I am wondering if you have given any though on supporting this es feature.

In the end it should somehow get merged into the array here.
https://github.com/babenkoivan/elastic-adapter/blob/1b8f417d17b6f32fc54b5bff57bab7108cb8e849/src/Documents/DocumentManager.php#L108

Cheers

**Comment by babenkoivan** [comment]
Hey @spiritinlife, would it be enough to add two methods in `SearchRequestBuilder`?

```php
// i.e. by using elastic-scout-driver-plus syntax 
Model::searchQuery($query)->preference($preference)->searchType($searchType)
```

We could also introduce a global default config for the `preference` or maybe a default value per model, what do you think?

**Comment by spiritinlife** [comment]
This fluent syntax could work great.

Maybe we should also introduce a method to override and add default chained operations per model that would be auto applied on every query instead of providing a config or a specific function override.

**Comment by babenkoivan** [comment]
@spiritinlife but you can already do that by overriding the `searchQuery` method, can't you?

```php
class MyModel extends Model
{
     use Searchable {
        searchQuery as baseSearchQuery;    
    }

     public static function searchQuery($query): SearchRequestBuilder
     {
         return static::baseSearchQuery($query)->preference($preference);
     }
}
```

**Comment by spiritinlife** [comment]
True, we can live with that. Although not the friendliest it can do :)

**Comment by babenkoivan** [comment]
Hey @spiritinlifem I've just released elastic-scout-driver-plus [v3.4.0](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v3.4.0) with `search_type` and `preference` support. Please check it out and feel free to reopen the issue if you have any troubles with it.
